### ISSUE TITLE
modify_services after produce flips push notifications service?

### ISSUE THREAD

**Issue by salmanasiddiqui** [fastlane/fastlane#14850]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Initialized app with `create_app_online` with push_notification service enabled. Created push certificate for the app using `pem`. Then using `modify_services` with push_notification service enabled causes the push notification service to be disabled on developer portal.

##### Command executed

```
bundle exec fastlane create_new_app
bundle exec fastlane build_ipa
```
```
lane :create_new_app do
  create_app_online(
      username: USERNAME,
      app_identifier: 'com.company.testapp3',
      app_name: "TestApp - Preview",
      company_name: 'TEAM NAME',
      app_version: '2019.2.0402',
      sku: 'com.company.testapp3'',
      team_id: 'QEQEW123',
      itc_team_name: 'TEAM NAME',
      enable_services: {
          game_center: 'on',
          in_app_purchase: 'on',
          push_notification: 'on',
          associated_domains: 'on'
      }
  )
  get_push_certificate(
      development: false,
      generate_p12: false,
      app_identifier: 'com.company.testapp3',
      username: USERNAME,
      team_id: 'QEQEW123',
      pem_name: 'testapp3',
      output_path: 'CERTS'
  )
end

lane :build_ipa do
  Spaceship::Tunes.login(USERNAME, ENV['FASTLANE_PASSWORD'])
  modify_services(
      username: USERNAME,
      app_identifier: 'com.company.testapp3',
      team_id: 'QEQEW123',
      services: {
          game_center: 'on',
          in_app_purchase: 'on',
          push_notification: 'on',
          associated_domains: 'on'
      }
  )
end
```
### Environment

<details>
  <pre> 
### Stack

| Key                         | Value                                              |
| --------------------------- | -------------------------------------------------- |
| OS                          | 10.14.4                                            |
| Ruby                        | 2.5.5                                              |
| Bundler?                    | true                                               |
| Git                         | git version 2.20.1 (Apple Git-117)                 |
| Installation Source         | ~/.rbenv/versions/2.5.5/bin/fastlane               |
| Host                        | Mac OS X 10.14.4 (18E226)                          |
| Ruby Lib Dir                | ~/.rbenv/versions/2.5.5/lib                        |
| OpenSSL Version             | OpenSSL 1.0.2r  26 Feb 2019                        |
| Is contained                | false                                              |
| Is homebrew                 | false                                              |
| Is installed via Fabric.app | false                                              |
| Xcode Path                  | /Applications/Xcode-10.2.1.app/Contents/Developer/ |
| Xcode Version               | 10.2.1                                             |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby   
opt_out_usage
import 'IosFastfile'

```
</details>
**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.122.0 | � Update available |

### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status |
| ----------------------- | ------- | ------------- |
| fastlane-plugin-appicon | 0.14.0  | ✅ Up-To-Date  |
| fastlane-plugin-brew    | 0.1.1   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.2.0        |
| bundler                   | 2.0.1        |
| etc                       | 1.0.0        |
| CFPropertyList            | 3.0.0        |
| concurrent-ruby           | 1.1.5        |
| i18n                      | 0.9.5        |
| minitest                  | 5.11.3       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.5        |
| activesupport             | 4.2.11.1     |
| public_suffix             | 2.0.5        |
| addressable               | 2.6.0        |
| ast                       | 2.4.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| mini_mime                 | 1.0.1        |
| mini_portile2             | 2.4.0        |
| nokogiri                  | 1.10.3       |
| rack                      | 2.0.7        |
| rack-test                 | 1.1.0        |
| xpath                     | 3.2.0        |
| capybara                  | 2.18.0       |
| claide                    | 1.0.2        |
| cliver                    | 0.3.2        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.6.1        |
| cocoapods-deintegrate     | 1.0.4        |
| cocoapods-downloader      | 1.2.2        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.1.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.1        |
| cocoapods-try             | 1.1.0        |
| colored2                  | 3.1.2        |
| escape                    | 0.0.4        |
| fourflusher               | 2.2.0        |
| gh_inspector              | 1.1.3        |
| molinillo                 | 0.6.6        |
| ruby-macho                | 1.4.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.9.0        |
| cocoapods                 | 1.6.1        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| connection_pool           | 2.2.2        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| dotenv                    | 2.7.2        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.64.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.4       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.5        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.13.1       |
| os                        | 1.0.1        |
| signet                    | 0.11.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2019.0331  |
| mime-types                | 3.2.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.3.0        |
| json                      | 2.2.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.1        |
| tty-spinner               | 0.9.0        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-appicon   | 0.14.0       |
| fastlane-plugin-brew      | 0.1.1        |
| git                       | 1.5.0        |
| jaro_winkler              | 1.5.2        |
| openssl                   | 2.1.2        |
| parallel                  | 1.17.0       |
| parser                    | 2.6.3.0      |
| websocket-extensions      | 0.1.3        |
| websocket-driver          | 0.7.0        |
| poltergeist               | 1.12.0       |
| rack-protection           | 2.0.5        |
| rainbow                   | 3.0.0        |
| redis                     | 4.1.1        |
| rest-client               | 2.0.2        |
| ruby-progressbar          | 1.10.0       |
| rubocop                   | 0.69.0       |
| sidekiq                   | 5.2.7        |
</details>
 </pre>
</details>

**Comment by janpio** [comment]
> First concern, `sync_code_signing` creates new provisioning profile after warning that existing one is not valid. On developer portal, the push notification service is not enabled.

So is there a bug in `sync_code_signing` that it first created a broken provisioning profile in step 1? Or is it correctly created first, but then messed up later?

> which calls `modify_services` with four services enabled

Why do you call that again if you just created the app with those settings? Does this actually change anything?

---

Can you include your `Fastfile` content in your initial issue please?

**Comment by salmanasiddiqui** [comment]
> So is there a bug in sync_code_signing that it first created a broken provisioning profile in step 1? Or is it correctly created first, but then messed up later?

sync_code_signing is working as expected

> Why do you call that again if you just created the app with those settings? Does this actually change anything?

we call `modify_services` in build ipa lane, so that if we change apple services we can do it for existing apps when we rollout next build. `modify_services` action is redundant only in the case when we need to initialize app and build ipa for it.
Seems to me that `modify_services` is causing the issue, it turns the push notification service off on developer portal cause of which new provisioning profile is created and then gym fails cause entitlements file has aps enabled.

**Comment by janpio** [comment]
> Seems to me that `modify_services` is causing the issue, it turns the push notification service off on developer portal

Could you please verify this? Then we have a proper bug that can be reported and looked into. Best create a new issue where you only report _this_ problem and all the necessary information to reproduce this.

**Comment by salmanasiddiqui** [comment]
@janpio I debugged it and I have found a minimal way to reproduce the problem.

1. create app with push_notification='on'
1. create push cert
1. use modify_services with push_notification='on'

`modify_services` for some reason turns push_notification off in this case. Updating the description of this issue with relevant info, also adding fastfile to reproduce the problem

**Comment by janpio** [comment]
(Could you possibly take a look @martinkasa? I remember you tried to fix that action in the last few weeks - you will have a much easier time to reproduce, understand and fix this.)

**Comment by martinkasa** [comment]
I tried to fix it. I found out that problem is probably wrong request created, but I do not know how to fix it.

What I did was, that I captured request by Charles proxy created by fastlane and tried to change values manually, but a service was never enabled.

Probably some API on apple portal was changed.

**Comment by janpio** [comment]
Ok, so `modify_services` probably doesn't do its job very well right now :/

**Comment by maxx-cyber** [comment]
I have the same issue.
It also revokes all APNs certificates associated with app id. It is very very critical for us. We have about 800 applications in more than 100 Apple Developer Program accounts and we have a lot of escalations from clients that push notifications stop working.
In addition, if there are no APNs certificates, it doesn't disable push_notification entitlement as expected. Also if set `push_notification` to `false` it also revokes APNs certificates.
The same problem is reproduced using `fastlane produce enable_services --push-notification` as reported but not fixed here https://github.com/fastlane/fastlane/issues/9131.
Also fastlane doesn't have ability to check is push_notification enabled on dev portal or not. So we don't have any workarounds for this problem.
I have latest fastlane version - 2.127.0
Please fix it! 🙏

**Comment by maxx-cyber** [comment]
I could not wait for the fix, so I've moved forward with workaround:
1. Perform `match` action and get updated provisioning profile at this point.
2. Check currently enabled services in this updated provisioning profile.
3. Then perform `produce enable_services` or `produce disable_services` only for those services  that need to be changed.
4. Perform `match` action again if needed.
But I doubt that original issue was fixed...
